### ISSUE TITLE
Zoom In feature!

### ISSUE THREAD

**Issue by wada3n** [phase1geo/Outliner#41]
Hi,
This is the kind of apps we really need in the Linux world.
I tip my hat to you :) 

What do you think if you add Zoom In feature.
by pressing a shortcut,Through it you can enter into one row and work in it only, to get more room on screen, and focus on the this row and its children.
With the addition of the path in the status bar, through it you can return step by step to the original outline.
This GIF shows the idea from the outlinely mac os app:
![May-19-2020 00-04-50](https://user-images.githubusercontent.com/13304792/82268972-2be07400-9968-11ea-986d-bb664a230eba.gif)

**Comment by phase1geo** [comment]
That's pretty cool. I'll see what I can come up with.

**Comment by phase1geo** [comment]
I have an initial implementation that provides for a "focus mode" by selecting a row and hitting the 'f' key. This will display the selected row and its descendants only. I don't currently have any support for something like the status bar.

The changes are in the development branch, if you want to give it a try.

**Comment by wada3n** [comment]
This is very beautiful.
I just have a suggestion, when entering more in depth of the outline.
Why when you hit Escape, it return once to the first outline, not in a sequence until you return to the beginning?
I think it would be better to return in a hierarchy to the beginning, just like entering

**Comment by phase1geo** [comment]
The current implementation doesn't have support for a stack of focus views. I need to think about how I want to implement that. The idea behind Escape is to get out of focus mode once you are in it. Consider it a work in progress.

I appreciate the feedback!

**Comment by wada3n** [comment]
@phase1geo Frankly, you are doing a great job, 
and in a short time, I could not imagine you doing all this.
I really appreciate the work you do.

**Comment by wada3n** [comment]
Hi @phase1geo 
I liked the information bar idea, it is very useful, because it supports smooth navigation back and forth.
I just think that show children should be set by default if focus mode is entered.
( tag filtring also )

![Peek 01-06-2020 16-59](https://user-images.githubusercontent.com/13304792/83428007-8b8b5480-a429-11ea-8e48-0c71b70132b1.gif)


As you can see, when any group is closed / hidden, everything disappears once if focus mode enters.

Practically, as long as I enter one of a row, I implicitly want the children of that row.

Also, in tag filtring, if the tags are in a row's child that doesn't appear until the triangle opens.

Thank you very much for this great job once again.

**Comment by wada3n** [comment]
Hi @phase1geo,
I like to sneak up on the new codes that you do on the project sometimes.

Yesterday I found the new idea you were applying about the focus pattern by making everyone dimmed except for the focused part.
I think the two ideas are kind of different and you could even be together in the application:
First, let's call it "branch it" mode instead of "Zoom in", which is a little similar to the implementation that I initially pursued with Path navigation bar, where you can easily switch between branches to give more space, And penetrate more and more into the tree hierarchy as needed!
The idea that it currently in implementation by obfuscating the unfocused portion is almost somewhat similar to the typewriter pattern and It would be great by the way.

I know the idea is in its early stages, but I think the two features could be together in this unique project.

**Comment by phase1geo** [comment]
I changed the implementation because it was tricky from an implementation perspective and because I don’t like it from a user perspective.

I don’t plan on supporting the idea of hiding rows in the application. It makes the code base much more complex and means that I have to add even more UI elements (like the navigation bar) to help the user manage the limited scope. This puts more onus on the user to learn how to use the application.

By “dimming” the uninteresting code, we give the user a way to focus on a portion of the document while still allowing them to navigate the rest of the document as they typically would. This is preferable from a code complexity and usage complexity perspective.

My plan is to place the top-most focused row at the top of the window when that row is given focus; however, I need to allow the document to scroll past the end of the text to help make that happen.

In addition, I also removed the ability to click on a tag to filter the document by that tag for the time being. I have a plan for how to make tags useful from a navigation perspective, but I don’t want to hold off on the 1.3 release.
